packages/generator/tests/acceptance/test_client_generator.py

```diff
@@ -72,6 +74,45 @@ def test___command_line_args___create_client_for_all_registered_measurements___r
     )
 
 
+@patch("pathlib.Path.cwd")
+def test___command_line_args_with_registered_measurements___create_client_using_interactive_mode___renders_without_error(
+    mock_directory_out: Mock,
+    test_assets_directory: pathlib.Path,
+    tmp_path_factory: pytest.TempPathFactory,
+    measurement_service: MeasurementService,
+) -> None:
+    temp_directory = tmp_path_factory.mktemp("measurement_plugin_client_files")
+    mock_directory_out.return_value = temp_directory
+    golden_path = test_assets_directory / "example_renders" / "measurement_plugin_client"
+    filename = "non_streaming_data_measurement_client.py"
+
+    with patch(
+        "click.prompt",
+        side_effect=[
+            "1",
+            "non_streaming_data_measurement_client",
+            "NonStreamingDataMeasurementClient",
+            "x",
+        ],
+    ):
+        with pytest.raises(SystemExit) as exc_info:
+            create_client(["--interactive"])
+
+    assert not exc_info.value.code
+    _assert_equal(
+        golden_path / filename,
+        temp_directory / filename,
+    )
+
+
+def test___command_line_args_without_registering_any_measurement___create_client_using_interactive_mode___raises_exception() -> (
+    None
+):
+    with pytest.raises(SystemExit):
+        with pytest.raises(click.ClickException):
```

Comment:
Why are you nesting pytest.raises?

Reply:
I have nested, to capture 2 exceptions. Now, I have updated the `systemExit` captures with `CLIRunner`, so these exceptions can be caught with that.